SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -885,6 +929,63 @@ Worker_ComponentUpdate USpatialSender::CreateUnreliableRPCUpdate(UObject* Target
 	return ComponentUpdate;
 }
 
+void USpatialSender::AddPendingUnreliableRPC(UObject* TargetObject, UFunction* Function, void* Parameters, Worker_ComponentId ComponentId, Schema_FieldId RPCIndex, const UObject*& OutUnresolvedObject)
+{
+	FUnrealObjectRef TargetObjectRef(PackageMap->GetUnrealObjectRefFromNetGUID(PackageMap->GetNetGUIDFromObject(TargetObject)));
+	if (TargetObjectRef == FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
+	{
+		OutUnresolvedObject = TargetObject;
+		return;
+	}
+
+	AActor* TargetActor = Cast<AActor>(PackageMap->GetObjectFromEntityId(TargetObjectRef.Entity).Get());
+	check(TargetActor != nullptr);
+	UNetConnection* OwningConnection = TargetActor->GetNetConnection();
+	if (OwningConnection == nullptr)
+	{
+		UE_LOG(LogSpatialSender, Warning, TEXT("AddPendingUnreliableRPC: No connection for object %s (actor %s, entity %lld)"), *TargetObject->GetName(), *TargetActor->GetName(), TargetObjectRef.Entity);
+		return;
+	}
+
+	APlayerController* Controller = Cast<APlayerController>(OwningConnection->OwningActor);
+	if (Controller == nullptr)
+	{
+		UE_LOG(LogSpatialSender, Warning, TEXT("AddPendingUnreliableRPC: Connection's owner is not a player controller for object %s (actor %s, entity %lld): connection owner %s"), *TargetObject->GetName(), *TargetActor->GetName(), TargetObjectRef.Entity, *OwningConnection->OwningActor->GetName());
+		return;
+	}
+
+	FUnrealObjectRef ControllerObjectRef = PackageMap->GetUnrealObjectRefFromObject(Controller);
+	if (ControllerObjectRef == FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
+	{
+		OutUnresolvedObject = Controller;
+		return;
+	}
+
+	TSet<TWeakObjectPtr<const UObject>> UnresolvedObjects;
+	FSpatialNetBitWriter PayloadWriter(PackageMap, UnresolvedObjects);
+
+	TSharedPtr<FRepLayout> RepLayout = NetDriver->GetFunctionRepLayout(Function);
+	RepLayout_SendPropertiesForRPC(*RepLayout, PayloadWriter, Parameters);
+
+	for (TWeakObjectPtr<const UObject> Object : UnresolvedObjects)
+	{
+		if (Object.IsValid())
+		{
+			// Take the first unresolved object
+			OutUnresolvedObject = Object.Get();
+			return;
+		}
+	}
+
+	FPendingUnreliableRPC RPC;
+	RPC.Offset = TargetObjectRef.Offset;
+	RPC.Index = RPCIndex;
+	RPC.Data.SetNumUninitialized(PayloadWriter.GetNumBytes());
+	FMemory::Memcpy(RPC.Data.GetData(), PayloadWriter.GetData(), PayloadWriter.GetNumBytes());
+	RPC.Entity = TargetObjectRef.Entity;
+	UnreliableRPCs.FindOrAdd(ControllerObjectRef.Entity).Add(RPC);
```

Comment:
performance nit:
Might be nice to implement move semantics for this operation
Or add an empty struct to the array, and set the object directly.